testing/test_config.py

```diff
@@ -1405,6 +1405,51 @@ def test_with_specific_inifile(
         assert inipath == p
         assert ini_config == {"x": "10"}
 
+    @pytest.mark.parametrize(
+        "tree,ini_file,args, expected_root",
+        [
+            pytest.param(
+                ["pytest.ini", "tests/test1"],
+                "cfg/cfg.ini",
+                ["tests"],
+                "",
+                id="root_ini_exist",
+            ),
+            pytest.param(
+                ["tests/a/test1"],
+                "tests/a/cfg.ini",
+                ["tests/a"],
+                "tests/a",
+                id="no_root_ini",
+            ),
+            pytest.param(
+                ["tests/a/test1", "tests/b/test2"],
+                "tests/a/cfg.ini",
+                ["tests"],
+                "tests/a",
+                id="nested",
+            ),
+        ],
+    )
+    def test_with_args_and_specific_inifile(
```

Comment:
Thanks for being through here, but I think we only need a test similar to the one posted in the original issue:

```python
def test_explicit_config_file_sets_rootdir(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> None:
    tests_dir = tmp_path / "tests"
    tests_dir.mkdir()

    monkeypatch.chdir(tmp_path)

    # No config file is explicitly given: rootdir is determined to be cwd.
    rootpath, found_inipath, *_ = determine_setup(None, [str(tests_dir)])
    assert rootpath == tmp_path
    assert found_inipath is None

    # Config file is explicitly given: rootdir is determined to be inifile's directory.
    inipath = (tmp_path / "pytest.ini")
    inipath.touch()
    rootpath, found_inipath, *_ = determine_setup(str(inipath), [str(tests_dir)])
    assert rootpath == tmp_path
    assert found_inipath == inipath
```

This should be enough to enforce the correct behavior.

Reply:
Much cleaner and on-point, replaced  `test_with_args_and_specific_inifile` with `test_explicit_config_file_sets_rootdir` thanks so much for clarification